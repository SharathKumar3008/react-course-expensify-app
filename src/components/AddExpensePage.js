import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { addExpense } from '../actions/expenses';
import expenses from '../tests/fixtures/expenses';

export class AddExpensePage extends React.Component{
    onSubmit = (expense) => {
        this.props.onSubmit(expense);
        this.props.history.push('/');
    }

    render(){
        return (
            <div>
                <ExpenseForm 
                    onSubmit = {this.onSubmit}
                />
            </div>
        );
    }
}

const mapDisptchToProps = (dispatch) => ({
    onSubmit : (expense) => dispatch(addExpense(expense))
});

export default connect(undefined, mapDisptchToProps)(AddExpensePage);