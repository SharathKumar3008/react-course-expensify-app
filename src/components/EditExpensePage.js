import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {editExpense, removeExpense} from '../actions/expenses';

/*const EditExpensePage = (props) => {
    console.log(props);
    return (
        <div>
            <ExpenseForm
                expense = {props.expense}
                onSubmit= {(expense)=>{
                    console.log('Updated',expense);
                    props.dispatch(editExpense({id: props.expense.id}, expense));
                    props.history.push('/');
                }}
            />
            <button onClick={()=>{
                props.dispatch(removeExpense({id :props.expense.id}));
                props.history.push('/');
            }}>Remove</button>
        </div>
    );
}*/

export class EditExpensePage extends React.Component{

    constructor(props) {
        super(props);
    }

    onSubmit = (expense) => {
        console.log(this.props.expense.id);
        console.log(expense);
        this.props.EditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    }

    onClick = () => {
        console.log(this.props.expense.id);
        this.props.RemoveExpense(this.props.expense.id);
        this.props.history.push('/');
    }
    
    render() {
        return (
            <div>
                <ExpenseForm
                    expense = {this.props.expense}
                    onSubmit= {this.onSubmit}
                />
                <button 
                    onClick={this.onClick}
                >Remove</button>
            </div>
        );
    }
}

const mapStatetoProps = (state, props) => {
    return {
        expense : state.expenses.find((expense)=>expense.id == props.match.params.id)
    };
};

const mapDisptchToProps = (dispatch) => ({
    EditExpense : (id, expense)=> dispatch(editExpense({id}, expense)),
    RemoveExpense : (id) => dispatch(removeExpense({id}))
});

export default connect(mapStatetoProps,mapDisptchToProps)(EditExpensePage);