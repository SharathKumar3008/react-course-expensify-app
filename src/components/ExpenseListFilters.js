import React from 'react';
import { connect } from 'react-redux';
import 'react-dates/initialize';
import { DateRangePicker } from 'react-dates';
//import 'react-dates/lib/css/_datepicker.css';
import {setTextFilter,sortByDate, sortByAmount, setStartDate, setEndDate} from '../actions/filters'; 


class ExpenseListFilters extends React.Component {
    state = {
        calendarfocused : null
    }

    onDatesChange = ({ startDate, endDate }) => {
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate));
    }

    onFocusChange = (calendarfocused) => {
        this.setState(()=>({calendarfocused}));
    }

    render() {
        return (
            <div>
                <input 
                    type='text' 
                    value={this.props.filters.text} 
                    onChange={(e)=>{
                        this.props.dispatch(setTextFilter(e.target.value));
                    }}
                />
                <select onChange={(e)=>{
                        e.target.value == 'date' ? this.props.dispatch(sortByDate())
                        : this.props.dispatch(sortByAmount())
                    }}
                >
                    <option value='date'>Date</option>
                    <option value='amount'>Amount</option>
                </select>
                <DateRangePicker
                    startDate ={this.props.filters.startDate}
                    startDateId='your_unique_start_date_id'
                    endDate={this.props.filters.endDate}
                    endDateId='your_unique_end_date_id'
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarfocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={(day)=>false}
                    showClearDates={true}              
                />
            </div>
        );
    }
}

const mapStatetoProps = (state) => {
    return {
        filters : state.filters
    }
}

export default connect(mapStatetoProps)(ExpenseListFilters);