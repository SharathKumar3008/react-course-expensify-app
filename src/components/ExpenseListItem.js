import React from 'react';
import { connect } from 'react-redux';
import {removeExpense} from '../actions/expenses';
import { Link } from 'react-router-dom';

export const ExpenseListItem = ({dispatch, id, description, amount, createdAt}) => (
    <tr>
        <td>
            <Link to={'edit/'+id}>
                {description}
            </Link>
        </td>
        <td>{amount}</td>
        <td>{createdAt}</td>
    </tr>
); 

export default connect()(ExpenseListItem);