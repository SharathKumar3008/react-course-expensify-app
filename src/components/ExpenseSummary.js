import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import ExpenseTotal from '../selectors/expenses-total';
import numeral from 'numeral'

export const ExpenseSummary = (props) => {
    const expenseWord = props.expensescount === 1 ? 'expense' : 'expenses';
    const formatNumeralAmount = numeral(props.expenseTotal/100).format('$0,0.00');
    return (
        <div>
            <h1>Viewing {props.expensescount} {expenseWord} totalling {formatNumeralAmount}</h1>
        </div>
    )
};

const mapStatetoProps = (state) => {
    const VisibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseTotal : ExpenseTotal(VisibleExpenses),
        expensescount: VisibleExpenses.length
    }
}
export default connect(mapStatetoProps)(ExpenseSummary);