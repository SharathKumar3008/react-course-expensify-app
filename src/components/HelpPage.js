import React from 'react';

const HelpPage = () => (
    <div>
        This is the Help Expense Component
    </div>
);

export default HelpPage;