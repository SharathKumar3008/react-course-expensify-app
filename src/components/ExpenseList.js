import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import SelectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
        <h1>This is Expense List</h1>
        <table>
            <thead>
                <tr>
                    <th>Description</th>
                    <th>Amount</th>
                    <th>Created At</th>
                </tr>
            </thead>
            <tbody>
                {
                    props.expenses.length === 0 ? 
                        <tr>
                            <td>No Data found</td>
                        </tr>
                    :
                    props.expenses.map((expense)=>
                        <ExpenseListItem key={expense.id} {...expense} />
                    )
                }
            </tbody>
        </table>

    </div>
);

const mapStatetoProps = (state) => {
    return {
        expenses : SelectExpenses(state.expenses, state.filters)
    }
}
export default connect(mapStatetoProps)(ExpenseList);