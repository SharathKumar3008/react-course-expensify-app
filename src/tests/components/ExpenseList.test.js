import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';

test('Should List Expenses correctly' , () => {
    const wapper = shallow(<ExpenseList expenses={expenses} />);
    expect(wapper).toMatchSnapshot();
});

test('Should Render Expense with empty array', () => {
    const wapper = shallow(<ExpenseList expenses={[]}  />);
    expect(wapper).toMatchSnapshot();
});