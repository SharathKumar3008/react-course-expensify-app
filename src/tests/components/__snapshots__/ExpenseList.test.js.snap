// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should List Expenses correctly 1`] = `
<div>
  <h1>
    This is Expense List
  </h1>
  <table>
    <thead>
      <tr>
        <th>
          Description
        </th>
        <th>
          Amount
        </th>
        <th>
          Created At
        </th>
      </tr>
    </thead>
    <tbody>
      <Connect(ExpenseListItem)
        amount={195}
        createdAt={0}
        description="Gum"
        id="1"
        key="1"
        note=""
      />
      <Connect(ExpenseListItem)
        amount={109500}
        createdAt={-345600000}
        description="Rent"
        id="2"
        key="2"
        note=""
      />
      <Connect(ExpenseListItem)
        amount={4500}
        createdAt={345600000}
        description="Credit Card"
        id="3"
        key="3"
        note=""
      />
    </tbody>
  </table>
</div>
`;

exports[`Should Render Expense with empty array 1`] = `
<div>
  <h1>
    This is Expense List
  </h1>
  <table>
    <thead>
      <tr>
        <th>
          Description
        </th>
        <th>
          Amount
        </th>
        <th>
          Created At
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          No Data found
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
